gpii/node_modules/dbOperation/src/DbDataStoreUtils.js

```diff
@@ -0,0 +1,545 @@
+ */
+gpii.dbOperation.dbDataStore.addGpiiKey = function (saveDataSource, gpiiKeyData) {
+    var promiseTogo = fluid.promise();
+    var data;
+
+    if (!gpiiKeyData) {
+        gpiiKeyData = {
+            prefsSafeId: null,
+            prefsSetId: null
+        };
+    }
+
+    data = {
+        schemaVersion: gpii.dbOperation.schemaVersion,
+        prefsSafeId: gpiiKeyData.prefsSafeId || null,
+        prefsSetId: gpiiKeyData.prefsSetId || null,
+        revoked: false,
+        revokedReason: null,
+        timestampCreated: gpii.dbOperation.getCurrentTimestamp(),
+        timestampUpdated: null
+    };
+
+    promiseTogo = gpii.dbOperation.dbDataStore.addRecord(saveDataSource, gpii.dbOperation.docTypes.gpiiKey, "id", data, gpiiKeyData.gpiiKey);
+
+    return promiseTogo;
+};
+
+/**
+ * Update a GPII key record
+ * @param saveDataSource {Component} The saveDataSource component provided by gpii.dbOperation.dbDataStore
+ * @param gpiiKey {String} The GPII key.
+ * @param gpiiKeyData {Object} The GPII key data. An example of the data:
+ *  {
+ *      type: {String},
+ *      schemaVersion: {String},
+ *      prefsSafeId: {String},
+ *      prefsSetId: {String},
+ *      revoked: {String},
+ *      revokedReason: {String},
+ *      timestampCreated: {Date},
+ *      timestampRevoked: {Date}
+ *  }
+ *
+ * @return {Promise} A promise object that carries either a response returned from CouchDB/PouchDB for updating the
+ * prefs safe record, or an error if `prefsSafeId` or `prefsSafeData` is not provided or `prefsSafeData.type` is not
+ * the doc type for prefsSafe.
+ */
+gpii.dbOperation.dbDataStore.updateGpiiKey = function (saveDataSource, gpiiKey, gpiiKeyData) {
+    var promiseTogo = fluid.promise();
+    var data, error;
+
+    if (!gpiiKey) {
+        error = gpii.dbOperation.composeError(gpii.dbOperation.errors.missingInput, {fieldName: "gpiiKey"});
+        promiseTogo.reject(error);
+    } else if (!gpiiKeyData) {
+        error = gpii.dbOperation.composeError(gpii.dbOperation.errors.missingDoc, {docName: gpii.dbOperation.docTypes.gpiiKey});
+        promiseTogo.reject(error);
+    } else {
+        data = {
+            type: gpiiKeyData.type,
+            schemaVersion: gpiiKeyData.schemaVersion,
+            prefsSafeId: gpiiKeyData.prefsSafeId,
+            prefsSetId: gpiiKeyData.prefsSetId,
+            revoked: gpiiKeyData.revoked,
+            revokedReason: gpiiKeyData.revokedReason,
+            timestampCreated: gpiiKeyData.timestampCreated,
+            timestampUpdated: gpii.dbOperation.getCurrentTimestamp(),
+            timestampRevoked: gpiiKeyData.timestampRevoked
+        };
+
+        promiseTogo = gpii.dbOperation.dbDataStore.updateRecord(saveDataSource, gpii.dbOperation.docTypes.gpiiKey, gpiiKey, data);
+    }
+    return promiseTogo;
+};
+
+// General Preferences Safes Functions
+// -----------------------------------
+
+/**
+ * Find a preferences by a GPII key
```

Comment:
This comment does not appear to match the function. The function does some transformation on the `data` argument but does not query the database (there is no "Find"). And it does not return a promise.